src/ApplicationLogs/LogReader.cs

```diff
@@ -51,161 +65,343 @@ protected override void OnSystemLoaded(NeoSystem system)
-                        catch (InvalidOperationException)
-                        {
-                            notification["state"] = "error: recursive reference";
-                        }
-                        return notification;
-                    }).ToArray();
-                    triggerList.Add(trigger);
                 }
-                blockJson["executions"] = triggerList.ToArray();
-                return blockJson;
+
+                return raw ?? JToken.Null;
+            }
+            else
+                throw new RpcException(-32602, "Invalid params");
+        }
+
+        #endregion
+
+        #region Console Commands
+
+        [ConsoleCommand("log block", Category = "ApplicationLog Commands")]
+        private void OnGetBlockCommand(UInt256 blockhash, string eventName = null)
+        {
+            var blockOnPersist = string.IsNullOrEmpty(eventName) ?
+                _neostore.GetBlockLog(blockhash, TriggerType.OnPersist) :
+                _neostore.GetBlockLog(blockhash, TriggerType.OnPersist, eventName);
+            var blockPostPersist = string.IsNullOrEmpty(eventName) ?
+                _neostore.GetBlockLog(blockhash, TriggerType.PostPersist) :
+                _neostore.GetBlockLog(blockhash, TriggerType.PostPersist, eventName);
+
+            if (blockOnPersist == null && blockOnPersist == null)
+                ConsoleHelper.Error($"No notify logs.");
+            if (blockOnPersist != null)
+                PrintExecutionToConsole(blockOnPersist);
+            if (blockPostPersist != null)
+            {
+                ConsoleHelper.Info("--------------------------------");
+                PrintExecutionToConsole(blockPostPersist);
             }
+        }
 
-            return null;
+        [ConsoleCommand("log tx", Category = "ApplicationLog Commands")]
+        private void OnGetTransactionCommand(UInt256 txhash, string eventName = null)
+        {
+            var txApplication = string.IsNullOrEmpty(eventName) ?
+                _neostore.GetTransactionLog(txhash) :
+                _neostore.GetTransactionLog(txhash, eventName);
+
+            if (txApplication == null)
+                ConsoleHelper.Error($"No notify logs.");
+            else
+                PrintExecutionToConsole(txApplication);
+        }
+
+        [ConsoleCommand("log contract", Category = "ApplicationLog Commands")]
+        private void OnGetContractCommand(UInt160 scripthash, uint page = 1, uint pageSize = 1, string eventName = null)
+        {
+            var txContract = string.IsNullOrEmpty(eventName) ?
+                _neostore.GetContractLog(scripthash, TriggerType.Application, page, pageSize) :
+                _neostore.GetContractLog(scripthash, TriggerType.Application, eventName, page, pageSize);
+
+            if (txContract.Count == 0)
+                ConsoleHelper.Error($"No notify logs.");
+            else
+                PrintEventModelToConsole(txContract);
         }
```

Comment:
I like this very much since it's convenient for developer to quick scan specific event log. Most of the function works well, but still somewhere has issues, for example:

```
neo> log contract 0xf0151f528127558851b39c2cd8aa47da7418ab28 0 3
Transaction Hash: 0x16890a963a9325bcce5fa874840ee9385fb52a81b08ba87a90cc0881842b03ad

  Event Name:  Transfer
  State Parameters:
    arg1: {"type":"ByteString","value":"Llmh1poEZKFMPOkBa0p1QYFoqys="}
    arg2: {"type":"ByteString","value":"352hGohJmxzp1lmMC322MRwTfag="}
    arg3: {"type":"Integer","value":"3339600000000"}
--------------------------------
Transaction Hash: 0x5f577775c8980b9d3f17e0c241a77c19232e39ff973fad9a4e3c854074c0f657

  Event Name:  Transfer
  State Parameters:
    arg1: {"type":"ByteString","value":"BrY0XpYy7nFjXOnPMsMBAxx5HRc="}
    arg2: {"type":"ByteString","value":"YeKHeeCXGEps8sSRlxbc4v97b1U="}
    arg3: {"type":"Integer","value":"128176221000"}
--------------------------------
neo> log contract 0xf0151f528127558851b39c2cd8aa47da7418ab28 1 3
Transaction Hash: 0x16890a963a9325bcce5fa874840ee9385fb52a81b08ba87a90cc0881842b03ad

  Event Name:  Transfer
  State Parameters:
    arg1: {"type":"ByteString","value":"Llmh1poEZKFMPOkBa0p1QYFoqys="}
    arg2: {"type":"ByteString","value":"352hGohJmxzp1lmMC322MRwTfag="}
    arg3: {"type":"Integer","value":"3339600000000"}
--------------------------------
Transaction Hash: 0x5f577775c8980b9d3f17e0c241a77c19232e39ff973fad9a4e3c854074c0f657

  Event Name:  Transfer
  State Parameters:
    arg1: {"type":"ByteString","value":"BrY0XpYy7nFjXOnPMsMBAxx5HRc="}
    arg2: {"type":"ByteString","value":"YeKHeeCXGEps8sSRlxbc4v97b1U="}
    arg3: {"type":"Integer","value":"128176221000"}
--------------------------------
Transaction Hash: 0x72426a7a73e5c9389d6385b5b1c6871558025e066eeda8cab1b21aeb59193257

  Event Name:  Transfer
  State Parameters:
    arg1: {"type":"ByteString","value":"qkj3lEssPRvcRa9lNHANUgZmNbI="}
    arg2: {"type":"ByteString","value":"Llmh1poEZKFMPOkBa0p1QYFoqys="}
    arg3: {"type":"Integer","value":"3339600000000"}
--------------------------------
```
I saw the default page index is `1`, but `0` still works. Maybe start from 0 makes more sense? And here I asked for 3 results in `size` param, but it returns `2` results at the beginning and returns `3` when I try for the second time. 
I even meet a situation like:
```
neo> log contract 0xf0151f528127558851b39c2cd8aa47da7418ab28 0 0
Error: No notify logs.
neo> log contract 0xf0151f528127558851b39c2cd8aa47da7418ab28 0 1
Error: No notify logs.
neo> log contract 0xf0151f528127558851b39c2cd8aa47da7418ab28 0 2
Transaction Hash: 0xd5756860d363680da5878f4f2bc5a5e8dd0d76bf5ab5cf8e570e5daf43a3cddd

  Event Name:  Transfer
  State Parameters:
    arg1: {"type":"ByteString","value":"uYxb4QZ8TaZy6W5rOQiLkLldUp4="}
    arg2: {"type":"ByteString","value":"oyJrAPWr9bXxnMN64ykd4lk6YKk="}
    arg3: {"type":"Integer","value":"18480412641180"}
--------------------------------
```

Reply:
This is because
```csharp
foreach (var (key, value) in _snapshot.Seek(prefixKey, SeekDirection.Backward)) // Get newest to oldest
{
    if (key.AsSpan().StartsWith(prefix))
    {
        var state = value.AsSerializable<ContractLogState>();
        if (state.Trigger == trigger)
        {
            if (index >= page && index < (pageSize + page)) // here this logic.
                yield return state;
            index++;
        }
    }
    else
        yield break;
}
```

But im fixing it to not allow 0 index for page and pagesize of 0